src/js/component.js

```diff
@@ -1425,16 +1425,20 @@ class Component {
    * Registers a component
    *
    * @param {String} name Name of the component to register
-   * @param {Object} comp The component to register
+   * @param {Object} Comp The component to register
    * @static
    * @method registerComponent
    */
-  static registerComponent(name, comp) {
-    if (!name) {
-      return;
+  static registerComponent(name, Comp) {
+    if (typeof name !== 'string' || !(/\S/).test(name)) {
```

Comment:
Sure. It may be overkill, though. I dunno. I am sometimes accused of being over-defensive. 😄 

It matches a string with any non-space character. The opposite of `\s`.